nodefs/bridge.go

```diff
@@ -38,11 +38,56 @@ type rawBridge struct {
 
 	files     []fileEntry
 	freeFiles []uint64
+
+	byFileID map[FileID]*Inode
+}
+
+// newInode creates creates new inode pointing to node.
+func (b *rawBridge) newInode(node Node, mode uint32, id FileID, persistent bool) *Inode {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if !id.Zero() {
+		// the same node can be looked up through 2 paths in parallel, eg.
+		//
+		//	    root
+		//	    /  \
+		//	  dir1 dir2
+		//	    \  /
+		//	    file
+		//
+		// dir1.Lookup("file") and dir2.Lookup("file") are executed
+		// simultaneously.  The matching FileIDs ensure that we return the
+		// same node.
+		old := b.byFileID[id]
```

Comment:
@navytux , @rfjakob  - what do you think of this approach?

Reply:
I like having the FileID table very much. Problem (maybe?) `node` gets thrown away. The FS may have stored data in it?

Reply:
I thought about splitting Lookup() into two phases, but that seems to just make things more complicated.

We can just document that `NewInode` may ignore `node` if it finds an older hard link.